src/Dscp/Educator/Block/Type.hs

```diff
@@ -0,0 +1,82 @@
+
+-- | Blocks- and blockchain-related datatypes definition.
+
+module Dscp.Educator.Block.Type
+       ( -- * Basic types
+         PrivateHeaderHash
+       , PrivateBlockHeader (..)
+       , pbhPrevBlock
+       , pbhBodyProof
+       , pbhAtgDelta
+       , PrivateBlockBody (..)
+       , pbbTxs
+       , PrivateBlock (..)
+       , pbHeader
+       , pbBody
+
+         -- * Constants
+       , genesisHeaderHash
+
+         -- * Helpers
+       , getPrevBlockRefMaybe
+       ) where
+
+import Control.Lens (makeLenses)
+import Dscp.Core.Types (ATGDelta (..))
+import Dscp.Crypto (Hash, unsafeHash)
+import Dscp.Crypto.MerkleTree (MerkleSignature)
+import Dscp.Educator.Txs (PrivateTx)
+
+----------------------------------------------------------
+-- Block elements
+----------------------------------------------------------
+
+-- | Hash of the private block.
+type PrivateHeaderHash = Hash PrivateBlockHeader
+
+-- | Header of a private block. There's no signatures here, as it's private anyway.
+-- During publishing, Educator will provide a signature as a part of transaction.
+data PrivateBlockHeader = PrivateBlockHeader
+    { _pbhPrevBlock :: !PrivateHeaderHash
+    -- ^ Previous header in the chain
+    , _pbhBodyProof :: !(MerkleSignature PrivateTx)
+    -- ^ Body payload proof (for now - only root of sized Merkle tree
+    -- over private transactions)
+    , _pbhAtgDelta  :: !ATGDelta
+    -- ^ Changes in courses taught by Educator
+    } deriving (Show, Eq, Generic)
+
+makeLenses ''PrivateBlockHeader
+
+-- | Genesis hash, serves as previous block reference for the first block.
+-- TODO: move to 'Genesis' module when it is formed somehow. Also, should
+-- private genesis hash actually make some sense?
+genesisHeaderHash :: PrivateHeaderHash
+genesisHeaderHash = unsafeHash ("pvaforever" :: ByteString)
+
+-- | Get previous block header, if previous block exists,
+-- 'Nothing' otherwise.
+getPrevBlockRefMaybe :: PrivateBlockHeader -> Maybe PrivateHeaderHash
+getPrevBlockRefMaybe PrivateBlockHeader {..} =
+    if _pbhPrevBlock == genesisHeaderHash
+    then Nothing
+    else Just _pbhPrevBlock
+
+-- | Private block body. Contains only private transactions (for now).
```

Comment:
>"for now"

Anything else is planned to be stored here? What am I missing?

Reply:
This is my comment, moved from the old file. Yes, cannot really think about anything else now, "for now" can be removed.